### PR TITLE
Add 2-approximation algorithm

### PR DESCRIPTION
**Pull request by Daer7**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -14,6 +15,10 @@ int main() {
   graph.display_graph();
 
   AlgorithmContext context{nullptr};
+  context.set_algorithm(new Aprox(graph));
+  context.run_algorithm();
+  context.set_algorithm(nullptr);
```
Comment (janwawruszczak1998) [inline main.cpp:20]: no need to setting strategy to nullptr, in next line setting ACO will clear it

File: Aprox.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+void Aprox::calculate_solution() {
+  const auto N = graph.get_order();
+  const auto &G = graph.get_graph();
+
+  std::vector<bool> vertex_included_in_solution(N, false);
+  for (unsigned i = 0; i < N; ++i) {
+    if (!vertex_included_in_solution[i]) {
+      for (unsigned j = 0; j < N; ++j) {
```
Comment (janwawruszczak1998) [inline Aprox.cpp:23]: 1. why casting from int to unsigned? use unsigned/auto j = 0u;
2. why not j = i + 1?

File: Aprox.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        if (auto are_adjacent = G[i][j].first;
+            are_adjacent && !vertex_included_in_solution[j]) {
+          vertex_included_in_solution[i] = true;
+          vertex_included_in_solution[j] = true;
```
Comment (janwawruszczak1998) [inline Aprox.cpp:27]: if you add two verticles  to solution, dont you have 2x bigger solution than is required?

File: Aprox.cpp (hunk 2 above)
Comment (Daer7) [reply Aprox.cpp:23]: 1. No casting involved, but gonna change to auto.
2. Good catch.

File: Aprox.cpp (hunk 3 above)
Comment (Daer7) [reply Aprox.cpp:27]: No, it's actually the right size because both added vertices won't be taken into consideration later in the algorithm.

**Pull request merged**